napari/layers/utils/text.py

```diff
@@ -1,242 +1,98 @@
+        is a format a string.
 
     Attributes
     ----------
-    text : array
-        the strings to be displayed
-    rotation : float
-        Angle of the text elements around the anchor point. Default value is 0.
-    anchor : str
+    values : np.ndarray
+        The text values to be displayed.
+    visible : bool
+        True if the text should be displayed, false otherwise.
+    size : float
+        Font size of the text, which must be positive. Default value is 12.
+    color : array
+        Font color for the text as an [R, G, B, A] array. Can also be expressed
+        as a string on construction or setting.
+    blending : Blending
+        The blending mode that determines how RGB and alpha values of the layer
+        visual get mixed. Allowed values are 'translucent' and 'additive'.
+        Note that 'opaque` blending is not allowed, as it colors the bounding box
+        surrounding the text, and if given, 'translucent' will be used instead.
+    anchor : Anchor
         The location of the text origin relative to the bounding box.
         Should be 'center', 'upper_left', 'upper_right', 'lower_left', or 'lower_right'.
     translation : np.ndarray
         Offset from the anchor point.
-    color : array
-        Font color for the text
-    size : float
-        Font size of the text. Default value is 12.
-    The blending mode that determines how RGB and
-        alpha values of the layer visual get mixed. Allowed values are
-        {'opaque', 'translucent', and 'additive'}. Note that 'opaque` blending
-        is not recommended, as colors the bounding box surrounding the text.
-    visible : bool
-        Set to true of the text should be displayed.
+    rotation : float
+        Angle of the text elements around the anchor point. Default value is 0.
     """
 
-    def __init__(
-        self,
-        text,
-        n_text,
-        properties={},
-        rotation=0,
-        translation=0,
-        anchor='center',
-        color='cyan',
-        size=12,
-        blending='translucent',
-        visible=True,
-    ):
-
-        self.events = EmitterGroup(
-            source=self,
-            auto_connect=True,
-            text=Event,
-            rotation=Event,
-            translation=Event,
-            anchor=Event,
-            color=Event,
-            size=Event,
-            blending=Event,
-            visible=Event,
-        )
-
-        self.events.block_all()
-        self._rotation = rotation
-        self._anchor = Anchor(anchor)
-        self._translation = translation
-        self._color = transform_color(color)[0]
-        self._size = size
-        self._blending = self._check_blending_mode(blending)
-        self._visible = visible
-
-        self._set_text(text, n_text, properties)
-        self.events.unblock_all()
-
-    @property
-    def values(self):
-        """np.ndarray: the text values to be displayed"""
-        return self._values
+    values: Array[str, (-1,)] = None
```

Comment:
```suggestion
    values: Array[str] = Field(default=[])
```

(and remove validator)

Reply:
also... the `(-1,)` here is useful if you need to enforce the shape of one axis, but not another, in a 2+D array... but for a vector, it's redundant.  (It get's used [here](https://github.com/napari/napari/blob/cd60cb66a12515b97f95ecb5365e9605d28ea6b1/napari/utils/events/custom_types.py#L22-L26))

Reply:
Thanks for in the info. I removed the `(-1,)` in values and translation.